### ISSUE TITLE
Wrong type response after overloading pipe transform

### ISSUE THREAD

**Issue by wambowams** [angular/vscode-ng-language-service#762]
**Describe the bug**

Using an overloaded transform method in a pipe isn't correctly recognized by the language service. It allways uses the first response type found, instead of the one matching the signature.

**To Reproduce**

I have a sorting-pipe looking similar to this:

```ts
  transform(values: TypeA[]): TypeA[];
  transform(values: TypeB[]): TypeB[];
  transform(values: TypeC[]): TypeC[];

  transform(values: TypeA[] | TypeB[] | TypeC[]): TypeA[] | TypeB[] | TypeC[] {
    /*
    sorting implementation
  */
  }
```
Using this pipe with values of TypeA works as expected, using it with values of TypeB or TypeC shows shows always TypeA as the response value.

**Expected behavior**

I expect the service to understand that the pipe is overloaded and display the right type and don't show an error if TypeA doesn't have a value from TypeB if the response should be TypeB in the first place.

**Additional context**

Versions:
angular: 9.1.0
angular-language-service: 0.901.4
IDE: vs-code

**Comment by ayazhafiz** [comment]
You're right, we always use the first signature found (https://github.com/angular/angular/blob/12fcc7cafe8bf6dd33416847327891b581281774/packages/language-service/src/typescript_symbols.ts#L225-L233):

```typescript
function selectSignature(type: ts.Type, context: TypeContext, _types: Symbol[]): Signature|
    undefined {
  // TODO: Do a better job of selecting the right signature. TypeScript does not currently support a
  // Type Relationship API (see https://github.com/angular/vscode-ng-language-service/issues/143).
  // Consider creating a TypeCheckBlock host in the language service that may also act as a
  // scratchpad for type comparisons.
  const signatures = type.getCallSignatures();
  return signatures.length ? new SignatureWrapper(signatures[0], context) : undefined;
}
```

We'll accept a patch for this use case, but in general we can't match signatures very well because TypeScript doesn't expose a convenient API for us to compare types. However, work is underway to move the language service to use the Ivy compiler as a backend, which typechecks templates with an instance of the TypeScript compiler -- with Ivy as a backend, this bug will be fixed automatically.

**Comment by ivanwonder** [comment]
https://github.com/microsoft/TypeScript/pull/33263/files#diff-c3ed224e4daa84352f7f1abcd23e8ccaR525-R527
Maybe this internal method can be used to select the right signature.

**Comment by ayazhafiz** [comment]
@ivanwonder yes it could be

**Comment by wambowams** [comment]
Thank u for your reply then I will be patient :D

**Comment by kyliau** [comment]
This has been fixed by the new Ivy-native language service, released in [v11.1.0](https://github.com/angular/vscode-ng-language-service/releases/tag/v11.1.0).
It's an opt-in feature for now, please give it a try and let us know if you have any feedback.
For the best editor experience, please make sure your project has [`strictTemplates`](https://angular.io/guide/angular-compiler-options#stricttemplates) enabled in `angularCompilerOptions`.
In case you run into similar bug, please file a new issue. I'll close this for now.
